### DISCUSSION TITLE
supabase.auth.signInWithOtp doesn't return any data

### DISCUSSION THREAD

**Discussion by ClineResearch** [supabase/supabase#14371] [Questions]
Hi,
Our project is with nuxt3/supabase.
We use supabase function auth.signInWithOtp for user authentification via a magic link.
How can we know if the user already exists when we call this function? How can we distinguish signin vs. registration? 
Thanks for your response.
JF

**Comment by GaryAustin1** [comment]
To know before the call you would have to have a rpc postgres function (security definer type) check if the email exists.

To know when the redirect occurs after the call if it is a new user, one way is to look in the user object at `last_sign_in_at` and `confirmed_at` to see if they are the same.   Maybe someone with a better idea than this will come along if the 2nd case is what you needed.

**Reply by ClineResearch** [reply]
Hi, thanks but I don't know how to implement this! I don't understand this given solution.

**Reply by GaryAustin1** [reply]
You did not respond to which way is your goal, before or after.

Before is just a postgres function to read auth.users given an email address passed in.
I just searched in the search tab for these discussions on `rpc call to see if email exists `
One hit that has a function example...  https://github.com/orgs/supabase/discussions/1282

After is using getSession to get the user info and compare the two values.
